File: String Rotation Game/2192.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int check_block(const vector<char> &s)
{
    if (s.empty())
        return 0;

    int total_current_block = 1;

    for (size_t i = 0; i < s.size() - 1; i++)
    {
        if (s[i] != s[i + 1])
            total_current_block++;
    }

    return total_current_block;
}

void solve()
{
    int n;
    cin >> n;

    string s;
    cin >> s;

    vector<char> s_vec(s.begin(), s.end());

    int max_block = 0;

    for (int j = 0; j < n; j++)
    {
        int current_block = check_block(s_vec);
        if (current_block > max_block)
            max_block = current_block;

        rotate(s_vec.rbegin(), s_vec.rbegin() + 1, s_vec.rend());
    }

    cout << max_block << "\n";
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
#endif

    int t;
    cin >> t;

    while (t--)
    {
        solve();
    }

    return 0;
}

File: String Rotation Game/2192.js
const fs = require("fs");

function solve() {
  const source = fs.existsSync("input.txt") ? "input.txt" : 0;
  const input = fs.readFileSync(source, "utf-8").trim().split(/\r?\n/);
  if (input.length === 0 || input[0] === "") return;
  let lineIndex = 0;
  const t = parseInt(input[lineIndex++], 10);

  function checkBlock(s) {
    let totalCurrentBlock = 0;

    for (let i = 0; i < s.length; i++) {
      if (s[i] !== s[i + 1]) totalCurrentBlock++;
    }

    return totalCurrentBlock;
  }

  for (let i = 0; i < t; i++) {
    const n = parseInt(input[lineIndex++], 10);
    let s = input[lineIndex++].split("");

    let maxBlocks = 0;

    for (let i = 0; i < n; i++) {
      let currentBlock = checkBlock(s);
      if (currentBlock > maxBlocks) maxBlocks = currentBlock;

      s.unshift(s[s.length - 1]);
      s.pop();
    }

    console.log(maxBlocks);
  }
}

solve();
